packages/orbit-components/src/Box/index.tsx

```diff
@@ -46,23 +45,319 @@ const Box = React.forwardRef<HTMLDivElement, Props>(
       children,
       dataTest,
       className,
-      ...smallMobile
+      display,
+      wrap,
+      direction,
+      position,
+      align,
+      justify,
+      textAlign,
+      elevation,
+      borderRadius,
+      overflow,
+      shrink,
+      grow,
+      zIndex,
+      width,
+      minWidth,
+      maxWidth,
+      height,
+      maxHeight,
+      top,
+      right,
+      bottom,
+      left,
+      color,
+      background,
+      padding,
+      margin,
     },
     ref,
   ) => {
-    const viewports = { smallMobile, mediumMobile, largeMobile, tablet, desktop, largeDesktop };
+    const viewportClasses = React.useMemo(() => {
```

Comment:
Don't you consider getting functions that we have in the initial implementation on search useful? It looks like it would make the component code clean. It's 233 lines of code compared to 363 here 🤔

Reply:
>  On Search it works because it never happens on their project.

That sounds strange :) How their setup is different? 

> Calling functions does not work if things change in runtime.

Depends on how you use them. Functions that are defined inside Box and Stack there work and will work for us too.

Reply:
> But calling a function in runtime to get a TW class from some prop value is the same as just building the class in runtime (with template strings, for example)

that's not the same, functions that return static objects like those defined in those components on Serch should work 🤔

Reply:
Their setup is not different. Their usage of the Box component is just in a way that props never change in runtime.

The functions do not return a static object. They return a string value based on the prop value. If that prop changes in runtime, the string will be different. And if the string contains a class that did not exist on build time, it won't apply the correct style because Tailwind did not consider the existence of such class.

Reply:
Okay, so this wont work then? 

```
const [isRed, setIsRed] = React.useState(false);

return (
  <>
  <Button onClick={() => setIsRed(prev => !prev)} />
  <Box  color={isRed ? "redNormal" : "blueNormal" />
  </>
)

// inside Box 
const getTailwindClass = (color: string, viewport: string) => {
  return viewport ? colorClasses[viewport][color] : colorClasses[color];
};

...
 className={cx(
   className,
   "orbit-box font-base box-border",
   color && getColorClasses(color),
   ..viewports.map(query => {
    if (!viewportSpecs[query]) return null;
    return getColorClasses(viewportSpecs[query]["color"], query);
)}
....
```

Reply:
If the classes mapped by `redNormal` and `blueNormal` do not exist anywhere else on the code, yes, it won't. Because tailwind won't generate the appropriate classes for them

Reply:
I thought (can't find that thread) that we already agreed that it's working when the function returns a complete string based on static objects (as I mentioned above). 

We knew that we can't construct strings dynamically via template literals, but when the ternary condition resolves to not dynamic string it's ok, the result of the function should be accessible at build time

You can also take a look at [example here](https://flaviocopes.com/tailwindcss-cannot-generate-classes-dynamically/)

@oreqizer can you take a look at that here? 🙏🏻

Reply:
https://tailwindcss.com/docs/content-configuration#dynamic-class-names


as long as classnames are unbroken strings, it **does not matter** where they're placed. whether in functions, objects, or just hovering in spaces. they just cannot be _constructed_ dynamically by slapping together multiple strings

Reply:
> whether in functions, objects, or just hovering in spaces

I don't see a reference to functions on the link you have sent. And from my experiments, it does not work with the Box implementation we have on Search

Reply:
Can you maybe share the link to the branch with your experiments? I think it's quite important to make sure about that since it will affect further migration and how we write them

Reply:
I stand corrected. As I was double-checking my example on the Search repo I noticed that the "problem" was actually on their config. They are missing tokens for classes that are defined on the Box implementation (again, it's not a problem because nobody is using them, that is why everything works fine there). But that is the ultimate reason why classes were not being generated.

Alright, I will refactor to function calls on our side 👍